common-items.lic

```diff
@@ -294,6 +294,18 @@ module DRCI
     end
   end
 
+  def have_item_by_look?(item, container)
+    item = "my #{item}" if item && !(item =~ /^my /i)
+    container = "my #{container}" if container && !(container =~ /^((in|on|under|behind|from) )?my /i)
+
+    case DRC.bput("look at #{item} in #{container}", 'You see nothing unusual', 'I could not find', 'What were you referring to')
```

Comment:
Are you suggesting the final command be `look at item from my container`? That doesn't make sense to me.

Reply:
Heh, the grammar part wasn't the objective of my suggestion. Sorry, I should have clarified. The change is to not hardcode the prepositions (in, from, under, ..) in the command so that the preposition optionally can be passed in as part of the argument.

Here's some examples of what I'm talking about:
```
DRCI.have_item_by_look?('rope', 'pack')         # look at my rope in my pack
DRCI.have_item_by_look?('rope', 'in drawer')    # look at my rope in drawer
DRCI.have_item_by_look?('rope', 'behind table') # look at my rope behind table
DRCI.have_item_by_look?('rope', 'on my shelf')  # look at my rope on my shelf
```

Does that make more sense?

As an aside, although in plain english the phrase "look at item from my container" doesn't read well, the game engine parses it fine.
```
> look my pouch from my pack
The soft gem pouch is made for easy storage of gems, and is used by gemshops everywhere to handle bulk transactions.
```

Thanks

Reply:
Another thought, to expand this method's utility to checking if an item is on your person (worn or stowed), then can modify the container argument in the signature to be `container = nil`.

Also, since this method uses the `look` verb, if an item has a descriptive look then it needs to handle more match strings. Since the permutations are boundless for what items LOOK like, but are fairly limited for when an item can't be found, I suggest this change:

```ruby
  def have_item_by_look?(item, container = nil)
    item = "my #{item}" if item && !(item =~ /^my /i)
    container = "in my #{container}" if container && !(container =~ /^((in|on|under|behind|from) )?my /i)

    case DRC.bput("look at #{item} #{container}", 'I could not find', 'What were you referring to', '.*')
    when 'I could not find', 'What were you referring to'
      false
    else
      true
    end
  end
```

Here's some quick tests:
```
>,e echo DRCI.have_item_by_look?('hairbrush')

[exec1]>look at my hairbrush 
I could not find what you were referring to.
> 
[exec1: false]

>,e echo DRCI.have_item_by_look?('soft pouch')

[exec1]>look at my soft pouch 
The soft gem pouch has been tied off.
The soft gem pouch is made for easy storage of gems, and is used by gemshops everywhere to handle bulk transactions.
> 
[exec1: true]

>,e echo DRCI.have_item_by_look?('pack')
[exec1]>look at my pack 
The steelsilk is woven through the fabric in such a way to form the image of a broad tower shield.
> 
[exec1: true]
```

Reply:
Another thought is use the existing `tap(item, container = nil)` method. Would that work for your use case and thus not need this new method?

Reply:
The thing about tap is that it's very RP breaking, in certain spots where you need to verify a lot of things at once. Like, it's a dead giveaway to me that someone else is using lich when they stand up, tap 5 things, and run out the room.

I like the silent 'look' verify where it works.

Reply:
That said, I'm all for switching to a negative match test, and fixing the adverbs like you suggested.

Reply:
One downside to .* matching is that it will match any game response, regardless of if it was actually the response to look, which could mean false positives on having things in noisy rooms. Thoughts?

Reply:
Yeah, that’s definitely a downside and entirely possible that an unrelated game response is received before a “What we’re you referring to?”

If the noun of the item you’re trying to look at will be in the LOOK then the .* maybe can be changed to something akin to `/\b#{item.strip.split(' ').last}\b/`